imagersite/imager_images/views.py

```diff
@@ -1,71 +1,139 @@
+                self.username and self.album.published != 'PUBLIC':
+            raise Http404('Album not found.')
 
-    if album.user.username != username and album.published != 'PUBLIC':
-        raise Http404('Album not found.')
+        return super().get(*args, **kwargs)
 
-    if album.user.username == username or album.published == 'PUBLIC':
-        context['album'] = album
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['album'] = self.album
+        return context
 
-    return render(request, 'imager_images/album.html', context)
 
+class AlbumGalleryView(ListView):
+    template_name = 'imager_images/album_gallery.html'
+    context_object_name = 'gallery'
 
-def album_gallery_view(request):
-    gallery = Album.objects.filter(published='PUBLIC')
-    context = {
-        'gallery': gallery,
-    }
-    return render(request, 'imager_images/album_gallery.html', context)
+    def get_queryset(self):
+        return Album.objects.filter(published='PUBLIC')
 
 
-def library_view(request):
-    username = request.user.get_username()
+class LibraryView(ListView):
+    template_name = 'imager_images/library.html'
+    model = Album
+    context_object_name = 'albums'
 
-    if username == '':
-        return redirect('auth_login')
+    def get_queryset(self):
+        return Album.objects.filter(user__username=self.username)
 
-    albums = Album.objects.all().filter(user__username=username)
-    photos = Photo.objects.filter(user__username=username)
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['photos'] = Photo.objects.filter(user__username=self.username)
+        return context
 
-    # if not owner:
-    #     albums = albums.all().filter(published='PUBLIC')
-    #     photos = photos.filter(published='PUBLIC')
+    def get(self, *args, **kwargs):
+        if not self.request.user.is_authenticated:
+            return redirect('auth_login')
 
-    context = {
-        'albums': albums,
-        'photos': photos,
-    }
-    return render(request, 'imager_images/library.html', context)
+        self.username = self.request.user.get_username()
+        return super().get(*args, **kwargs)
+
+
+class LibraryAddView(CreateView):
+    """
+    This creates a view that PhotoAddView and AlbumAddView inherit from.
+    """
+    success_url = '../library'
+
+    def get(self, *args, **kwargs):
+        if not self.request.user.is_authenticated:
+            return redirect('auth_login')
+    
+        return super().get(*args, **kwargs)
+
+    def post(self, *args, **kwargs):
+        if not self.request.user.is_authenticated:
+            return redirect('auth_login')
+    
+        return super().post(*args, **kwargs)
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs.update({'username': self.request.user.username})
+        return kwargs
```

Comment:
Since, you only need to do this on the `AlbumAddView`, this should probably not be in this base `LibraryAddView` class.